Common/include/linear_algebra/CSysMatrix.hpp

```diff
@@ -78,13 +78,39 @@ struct mkl_jit_wrapper<float> {
 class CConfig;
 class CGeometry;
 
+struct CSysMatrixComms {
+  /*!
+   * \brief Routine to load a vector quantity into the data structures for MPI point-to-point
+   *        communication and to launch non-blocking sends and recvs.
```

Comment:
I also moved the MPI comms of CSysVectors out of the matrix, in time we can make this even more generic to communicate anything that "looks like" a CSysVector which would allow us to communicate solution variables without having to define enum types for them.